mobile/native/src/ln_dlc/mod.rs

```diff
@@ -544,13 +537,23 @@ fn derive_trades_from_filled_orders() -> Result<Vec<WalletHistoryItem>> {
                 .trader_margin()
                 .expect("Filled order to have a margin");
 
+            let execution_price = Decimal::try_from(
+                first
+                    .execution_price()
+                    .expect("execution price to be set on a filled order"),
```

Comment:
You could just return an error here instead of panicking as this function already returns an error.